### PR TITLE
Channels now starting from 1 & Alternating row colors

### PR DESCRIPTION
**Pull request by jandegen**
image

The bottom controll panel is now also scrollable

### CODE REVIEW

**Review by nicolaskolbenschlag — CHANGES_REQUESTED** [review]
With a light theme, the table looks very dark. I personally would prefer a white and light-gray stripes instead of gray and dark-gray. Also the font-color should be updated because black on gray is not so easy to read.
image

File: Apps/frontend/src/app.scss [hunk 1]
```diff
@@ -179,33 +190,47 @@ nav {
     }
 
     & > table {
+      display: flex;
```
Comment (PhlppKrmr) [inline Apps/frontend/src/app.scss:193]: All the scaling stuff is kinda useless cause I already fixed it in #171 + I don't really like the way the bottom control panel scrolls independently from the rest of the screen (you can basically scroll below the control panel when >100% zooming; see image below)
image

Comment (PhlppKrmr) [reply Apps/frontend/src/app.scss:193]: I'll make sure to look into your styling changes when merging your and #171's branch, but I think especially the responsiveness is not handled perfectly here.

File: Apps/frontend/src/components/Slider.svelte [hunk 2]
```diff
@@ -14,6 +14,6 @@
 <div class={className} data-cy={dataCy}>
   <input {id} type="range" bind:value on:input={onInput} {min} {max} {step} />
   {#if calculateDisplayedValue !== undefined}
-    <span>{calculateDisplayedValue(value)}</span>
+    <div style="display: inline;">{calculateDisplayedValue(value)}</div>
```
Comment (PhlppKrmr) [inline Apps/frontend/src/components/Slider.svelte:17]: Please no inline styling, all styling should be in `app.scss`. On the same note, why did you change `span` to `div`?

File: Apps/frontend/src/const.js [hunk 3]
```diff
@@ -41,7 +41,7 @@ export const MIN_AMPLITUDE = 0.0;
 export const MAX_AMPLITUDE = NUM_INTERVALS_HORIZONTAL / 2;
 export const WAVE_CURSOR_SIZE = 50;
 
-export const MIN_CONTROL_PANEL_BOTTOM_HEIGHT = 280; //px
+export const MIN_CONTROL_PANEL_BOTTOM_HEIGHT = 150; //px
```
Comment (PhlppKrmr) [inline Apps/frontend/src/const.js:44]: At least for me the accordion button does no longer appear at the correct moment when zooming is changed due to this change.

File: Apps/frontend/src/views/ControlPanelBottom.svelte [hunk 4]
```diff
@@ -20,81 +20,88 @@
...
+  <thead>
+    <th>
+      {#if $expandedPanelOpen && controlPanelBottomHeight <= MIN_CONTROL_PANEL_BOTTOM_HEIGHT}
+        <button
+          class="icon-button icon-button--small mui-icon--close"
+          on:click={() => ($expandedPanelOpen = false)}
+          data-cy="expanded-control-panel-close-button"
         />
-      </td>
+      {:else}
```
Comment (PhlppKrmr) [inline Apps/frontend/src/views/ControlPanelBottom.svelte:33]: I like the idea of adding "Channel" here 👍

File: Apps/frontend/src/views/ControlPanelBottom.svelte [hunk 5]
```diff
@@ -20,81 +20,88 @@
...
-      </td>
-      <td>
-        <Slider
-          className="control-panel--entry"
-          bind:value={$offsetAdjustment[index]}
-          min={-1.0}
-          max={1.0}
-          step={0.01}
-          dataCy={`offsetSlider-${index}`}
+  <thead>
```
Comment (PhlppKrmr) [inline Apps/frontend/src/views/ControlPanelBottom.svelte:25]: Did you introduce `thead` (and `tbody`) for ease of styling in `app.scss`?

File: Apps/frontend/src/views/ControlPanelBottom.svelte [hunk 6]
```diff
@@ -20,81 +20,88 @@
...
+    <tr>
+      <td > Common </td>
+      <td><!--Placeholder--></td>
+      <td><!--Placeholder--></td>
+      <td><!--Placeholder--></td>
       <td>
         <Slider
           className="control-panel--entry"
-          bind:value={$timeSweep[index]}
+          onInput={() => {
```
Comment (PhlppKrmr) [inline Apps/frontend/src/views/ControlPanelBottom.svelte:52]: Can you add a comment indicating that this is the common slider? Sth. like `<!--Common TimeSweep Slider-->` right above `<Slider`

File: Apps/frontend/src/events.js [hunk 7]
```diff
@@ -1,2 +1,2 @@
 export const EVENT_LOADED_CHANNEL_CONFIG =
-  "event_loaded_channel_config_from_rest";
+  "event_loaded_channel_config_from_rest";
```
Comment (PhlppKrmr) [inline Apps/frontend/src/events.js:2]: What did even change?

File: Apps/frontend/src/app.scss [hunk 8]
```diff
@@ -179,33 +190,47 @@ nav {
...
+    & > table tr:nth-child(odd) {
+      background-color: #999999;
     }
   }
 
   &--entry {
-    & > span {
-      position: absolute;
-    }
+    & > input {
```
Comment (PhlppKrmr) [inline Apps/frontend/src/app.scss:231]: For what do we need `input` here? And why remove the `span: absolute` and then change it to an `inline div`?

File: Apps/frontend/src/app.scss [hunk 9]
```diff
@@ -78,8 +78,18 @@ nav {
   .icon-button {
     filter: invert(0) !important;
   }
+
+  table tr:nth-child(even) {
```
Comment (PhlppKrmr) [inline Apps/frontend/src/app.scss:82]: Really like the idea of zebra colouring, but I would probably recomment alternating between light grey and white for light mode and [background-color == #242424] and a slightly brighter grey tone (maybe sth. like #404040.

File: Apps/frontend/cypress/e2e/4-control-panel-bottom/amplitude-slider.cy.js [hunk 10]
```diff
@@ -12,14 +12,14 @@ describe("amplitudeSlider", () => {
   it(`has label that displays reciprocal value "Infinity"`, () => {
     cy.get(`[data-cy="amplitudeSlider-0"]`).within(() => {
       cy.get('[type="range"]').invoke("val", 2).trigger("change");
-      cy.get("span").should("have.text", "0.50");
+      cy.get("div").should("have.text", "0.50");
```
Comment (PhlppKrmr) [inline Apps/frontend/cypress/e2e/4-control-panel-bottom/amplitude-slider.cy.js:15]: Cross-dependent with a few comments further down.

**Review by PhlppKrmr — CHANGES_REQUESTED** [review]
Have some questions and am not quite happy with how the site scales. We could try merging #171 in the near future and then adapt the styling here to still work with responsiveness goals.

File: Apps/frontend/src/app.scss (hunk 1 above)
Comment (jandegen) [reply Apps/frontend/src/app.scss:193]: We will use yours as discussed

File: Apps/frontend/src/views/ControlPanelBottom.svelte (hunk 5 above)
Comment (jandegen) [reply Apps/frontend/src/views/ControlPanelBottom.svelte:25]: Is also used for styling

File: Apps/frontend/src/events.js (hunk 7 above)
Comment (jandegen) [reply Apps/frontend/src/events.js:2]: Nothing, maybe linting removed whitespaces or so

File: Apps/frontend/src/app.scss (hunk 8 above)
Comment (jandegen) [reply Apps/frontend/src/app.scss:231]: We don't have any spans inside the source code anymore. span caused a weird behavior so I changed it to div

File: Apps/frontend/cypress/e2e/4-control-panel-bottom/amplitude-slider.cy.js (hunk 10 above)
Comment (jandegen) [reply Apps/frontend/cypress/e2e/4-control-panel-bottom/amplitude-slider.cy.js:15]: As described changed to missbehavoir

File: Apps/frontend/src/const.js (hunk 3 above)
Comment (jandegen) [reply Apps/frontend/src/const.js:44]: Reversed

**Comment by jandegen** [comment]
> With a light theme, the table looks very dark. I personally would prefer a white and light-gray stripes instead of gray and dark-gray. Also the font-color should be updated because black on gray is not so easy to read. image

Now fixed with :

image

File: Apps/frontend/src/app.scss (hunk 9 above)
Comment (jandegen) [reply Apps/frontend/src/app.scss:82]: image
 What about this?

File: Apps/frontend/src/components/Slider.svelte (hunk 2 above)
Comment (jandegen) [reply Apps/frontend/src/components/Slider.svelte:17]: Moved to css class

File: Apps/frontend/src/app.scss (hunk 9 above)
Comment (PhlppKrmr) [reply Apps/frontend/src/app.scss:82]: Definitely better. Can we use the respective main background colour (white and #242424) + light grey (for light mode) and dark grey (for dark mode), do you think that'd look good?

Comment (jenswaechtler) [reply Apps/frontend/src/app.scss:82]: Personally, I think it's actually okay that way.

File: Apps/frontend/src/events.js (hunk 7 above)
Comment (jenswaechtler) [reply Apps/frontend/src/events.js:2]: `npm run format` causes for me that the whitespace is inserted again. Try running it again on your end.

**Review by jenswaechtler — CHANGES_REQUESTED** [review]

File: Apps/frontend/src/app.scss (hunk 9 above)
Comment (jandegen) [reply Apps/frontend/src/app.scss:82]: image
 #242424 is to dark in my opinion so

#404040 is better I think
image

Light mode is now:
image

Comment (PhlppKrmr) [reply Apps/frontend/src/app.scss:82]: Your light mode is now fine with me, the lighter colour of the dark mode is a little too hefty imo, would prefer a little darker.

Fyi, my idea was for it to look sth. like this:
- Dark mode:
image

-Light mode:
image

Comment (PhlppKrmr) [reply Apps/frontend/src/app.scss:82]: Would have my proposal ready, shall I commit?

Comment (jandegen) [reply Apps/frontend/src/app.scss:82]: Yes just add it
